nodejs/awsx/ec2/vpc.ts

```diff
@@ -83,13 +83,74 @@ export class Vpc extends pulumi.ComponentResource {
             }, { parent: this, aliases: [{ parent: pulumi.rootStackResource }] });
             this.id = this.vpc.id;
 
+            this.createSubnets(
+                cidrBlock, numberOfAvailabilityZones,
+                assignGeneratedIpv6CidrBlock, args.subnets || [
+                    { type: "public" },
+                    { type: "private" },
+                ], opts);
+
+            // Create an internet gateway if we have public subnets.
+            this.addInternetGateway(name, this.publicSubnets);
+
+            // Create nat gateways if we have private subnets.
+            createNatGateways(name, this, numberOfAvailabilityZones, numberOfNatGateways);
+        }
+
+        this.registerOutputs();
```

Comment:
the code from `// create an internet gateway` to this line are all the same as before.